exes/mqtt-broadcaster/inc/mqtt_broadcaster.hpp

```diff
@@ -0,0 +1,560 @@
+#pragma once
+
+#include <sparkplug_b/sparkplug_b.pb.h>
+#include <async_mqtt/all.hpp>
+#include <boost/asio.hpp>
+
+#include <tfc/dbus/string_maker.hpp>
+
+#include "config.hpp"
+
+using org::eclipse::tahu::protobuf::Payload;
+using org::eclipse::tahu::protobuf::Payload_Metric;
+
+namespace asio = boost::asio;
+namespace details = tfc::ipc::details;
+
+class network_manager {
+public:
+  auto connect_socket(auto&& socket, auto&& resolved_ip) -> asio::awaitable<void> {
+    co_await asio::async_connect(socket, resolved_ip, asio::use_awaitable);
+  }
+
+  auto set_sni_hostname(auto&& native_handle, std::string broker_address) -> void {
```

Comment:
passing std::string by value is not preferred even though the string is small, change to `std::string const& broker_address`